File: admin/src/main/java/com/project/admin/service/dto/techblog/TechBlogDetailResult.java
package com.project.admin.service.dto.techblog;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.project.admin.constant.Region;
import com.project.admin.constant.Source;
import com.project.admin.constant.Status;
import com.project.admin.domain.entity.TechBlog;
import com.project.admin.domain.entity.TechBlogApproval;
import java.time.LocalDateTime;
import java.util.List;

public record TechBlogDetailResult(
        Long id,
        Source source,
        Region region,
        String title,
        String url,
        @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Seoul")
        LocalDateTime publishedAt,
        String publisher,
        Status status,
        String summaries,
        List<String> tags) {

    public static TechBlogDetailResult from(TechBlog techBlog, TechBlogApproval approval) {

        return new TechBlogDetailResult(
                techBlog.getId(),
                techBlog.getSource(),
                techBlog.getRegion(),
                techBlog.getTitle(),
                techBlog.getUrl(),
                techBlog.getPublishedAt(),
                approval != null ? String.valueOf(approval.getUpdatedBy()) : null,
                techBlog.getStatus(),
                approval != null ? approval.getSummary() : null,
                techBlog.getTags()

        );
    }

    ;
}

File: api/src/main/kotlin/com/project/api/service/dto/TechBlogDetailResult.kt
package com.project.api.service.dto

import com.project.api.constants.Region
import com.project.api.constants.Source
import com.project.api.entity.TechBlog
import java.time.LocalDateTime

data class TechBlogDetailResult(
    val id: Long,
    val source: Source,
    val region: Region,
    val title: String,
    val url: String,
    val tags: List<String>,
    val publishedAt: LocalDateTime,
    val summary: String?,
) {
    companion object {
        fun from(techBlog: TechBlog) = TechBlogDetailResult (
            id =techBlog.id,
            source = techBlog.source,
            region = techBlog.region,
            title = techBlog.title,
            url = techBlog.url,
            publishedAt = techBlog.publishedAt,
            tags = techBlog.tags ?: emptyList(),
            summary =  techBlog.approval?.summary
        )

    }
}
